### DISCUSSION TITLE
Can Elsa be restarted without restarting Host?

### DISCUSSION THREAD

**Discussion by tomy2105** [elsa-workflows/elsa-core#1501] [Q&A]
I'm using Elsa 2.0 and ASP.NET.

Can Elsa be restarted without restaring Host?
I ask because at Elsa startup I create some activities via IActivityTypeProvider.
Now I need these activities to be "refreshed" from time to time (some properties removes, some added)....
I currently need to restart whole ASP.NET stack in order to do so.
Can this be done differently without need to restart ASP.NET?
Some method to force "refresh" of Elsa Activities created using IActivityTypeProvider?

**Comment by sfmskywalker** [comment]
That's certainly possible.

There's a service called `ActivityTypeService` that caches all `ActivityType` objects provided by all registered `IActivityTypeProvider` implementations. 

If you evict this cache, all activity type providers will be invoked again.

To evict the cache, inject `ICacheSignal` into your class and do the following:

```csharp
await _cacheSignal.TriggerAsync(ActivityTypeService.CacheKey);
```

That will evict the activity type cache on all nodes in your cluster (provided that you configured a [distributed cache signal provider](https://elsa-workflows.github.io/elsa-core/docs/next/hosting/hosting-distributed-hosting#distributed-cache-signal-provider)).

**Reply by tomy2105** [reply]
Is there a way to get when the cache was filled last time?

I ask because if you have a cluster with multiple nodes who evict cache every hour you will get cache evicted n times per hour (n is number of nodes in cluster). 
Is there a way add another listener to this cache signal to update internal time when next eviction needs to take place, or something similar?
